src/language/builtins/BuiltinsList.re

```diff
@@ -0,0 +1,2844 @@
+                        ),
+                        (
+                          Pat.cons(
+                            Pat.var("x"),
+                            Pat.cons(Pat.var("y"), Pat.var("ys")),
+                          ),
+                          let_(
+                            Pat.tuple([Pat.var("xs"), Pat.var("ys")]),
+                            ap(Forward, var("split"), var("ys")),
+                            tuple([
+                              cons(var("x"), var("xs")),
+                              cons(var("y"), var("ys")),
+                            ]),
+                          ),
+                        ),
+                      ],
+                    ),
+                    None,
+                    None,
+                  ),
+                  None,
+                ),
+                let_(
+                  Pat.var("merge_sort"),
+                  fix_f(
+                    Pat.var("merge_sort"),
+                    fn(
+                      Pat.var("xs"),
+                      match(
+                        var("xs"),
+                        [
+                          (Pat.list_lit([]), list_lit([])),
+                          (
+                            Pat.list_lit([Pat.var("x")]),
+                            list_lit([var("x")]),
+                          ),
+                          (
+                            Pat.wild(),
+                            let_(
+                              Pat.tuple([
+                                Pat.var("left"),
+                                Pat.var("right"),
+                              ]),
+                              ap(Forward, var("split"), var("xs")),
+                              ap(
+                                Forward,
+                                var("merge"),
+                                tuple([
+                                  var("cmp"),
+                                  ap(
+                                    Forward,
+                                    var("merge_sort"),
+                                    var("left"),
+                                  ),
+                                  ap(
+                                    Forward,
+                                    var("merge_sort"),
+                                    var("right"),
+                                  ),
+                                ]),
+                              ),
+                            ),
+                          ),
+                        ],
+                      ),
+                      None,
+                      None,
+                    ),
+                    None,
+                  ),
+                  ap(Forward, var("merge_sort"), var("xs")),
+                ),
+              ),
+            ),
+            None,
+            None,
+          )
+        )
+      );
+    },
+  },
+  {
+    str: {|fix slice -> fun (start, len, xs) -> take(drop(xs, start),len)|},
```

Comment:
We should consider if we want end or length on slice. Python/js do end index but I could see it either way.